### ISSUE TITLE
Species taxonomic status in WFO differs from status according to WFO as returned by TNRS

### ISSUE THREAD

**Issue by ojalaquellueva** [ojalaquellueva/TNRSbatch#5]
Reported by Christopher Johnson (c.johnson AT unibas.ch):

Entering "Ranunculus glacialis L." on TNRS with family classification = WFO and only WFO checked yields "Synonym" with the accepted name "Beckwithia glacialis". However, entering "Ranunculus glacialis L." on WFO directly shows that it is an accepted name.

Conversely, entering "Beckwithia glacialis" on TNRS with family classification = WFO and only WFO checked yields "Accepted". However, entering "Beckwithia glacialis" on WFO directly shows that it is a synonym of "Ranunculus glacialis L.".

**Comment by ojalaquellueva** [comment]
Issue replicated. Consulted TNRS via web interface: [https://tnrs.biendata.org](https://tnrs.biendata.org). Consulted WFO via url: [http://www.worldfloraonline.org/](http://www.worldfloraonline.org/).

**Comment by ojalaquellueva** [comment]
Not a bug. Taxonomic status returned by TNRS for species "Ranunculus glacialis L." accurately reflects content of WFO data extract (v.2019.05) as imported to the TNRS on 17 Jul 2020:

```
cd /home/bien/tnrs/data/db/wfo
grep "Ranunculus glacialis" classification.txt
```

Returns:

```
wfo-0000461034	217323-2	Ranunculus glacialis	SPECIES		L.	Ranunculaceae	Ranunculus	glacialis				Sp. Pl. 553 1753	Synonym	wfo-0000561740		2012-03-23	2012-03-23	http://www.theplantlist.org/tpl1.1/record/kew-2524023
wfo-0000461035	77069401-1	Ranunculus glacialis var. gelidus	VARIETY		(Kar. & Kir.) Finet & Gagnep.	Ranunculaceae	Ranunculus	glacialis	gelidus	var.		Bull. Soc. Bot. France 51: 307 1904	Synonym	wfo-0000460511		2012-03-23	2012-03-23	http://www.theplantlist.org/tpl1.1/record/kew-2524024
wfo-0000464756	217325-2	Ranunculus glacialis var. chamissonis	VARIETY		(Schltdl.) L.D.Benson	Ranunculaceae	Ranunculus	glacialis	chamissonis	var.		Amer. Midl. Naturalist 40: 226 1948	Synonym	wfo-0000561741		2012-03-23	2012-03-23	http://www.theplantlist.org/tpl1.1/record/kew-2529101
wfo-0000742302	217324-2	Ranunculus glacialis subsp. camissonis	SUBSPECIES		(Schltdl.) Hultén	Ranunculaceae	Ranunculus	glacialis	camissonis	subsp.		Acta Univ. Lund. n.f., 40(1): 753 1944	Synonym	wfo-0000561741		2012-03-23	2012-03-23	http://www.theplantlist.org/tpl1.1/record/kew-2891178
```

Note wfo-0000461034 "Ranunculus glacialis L." listed as synonym of wfo-0000561740. Grepping wfo-0000561740:

```
grep wfo-0000561740 classification.txt
```

references (among other lines):

```
wfo-0000561740	30240249-2	Beckwithia glacialis	SPECIES	wfo-4000004296	(L.) Á.Löve & D.Löve	Ranunculaceae	Beckwithia	glacialis				Acta Horti Gothob. 20: 142 1956	Accepted			2012-03-23	2012-03-23	http://www.theplantlist.org/tpl1.1/record/kew-2672304
```

**Comment by ojalaquellueva** [comment]
Not a bug. Difference is due to TNRS database being out of sync with source database. All source data is stored locally within the TNRS; sources are not consulted in real time. Local caching is a design feature that provides much faster performance and enables restructuring of source data to a common schema that better supports key TNRS features. Infrequent updates of the TNRS is a deliberate policy that enables archiving of all previous versions of the TNRS for reproducibility. However, increased frequency of updates (up to 4x / yr) should be possible and reduce the degree to which the TNRS gets out of sync with its sources.

**Comment by ojalaquellueva** [comment]
Changing label from Bug to Enhancement (=feature request). Enhancement would be greater frequency of updates, to about 4X per year. This will require some coding to ensure that all access points (API, R package and web interface) can be updated automatically and quickly, without the need for extensive manual interventions currently required.

**Comment by ojalaquellueva** [comment]
Short term fix: update WFO content in the TNRS database with WFO v.2022.07.

**Comment by ojalaquellueva** [comment]
Issue resolved with update of TNRS to app version 5.2, database version 4.4 on 2024-01-17. TNRS now uses WFO version v.2023.06.
